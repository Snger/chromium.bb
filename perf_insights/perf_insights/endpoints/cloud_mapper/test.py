# Copyright (c) 2015 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.
import os
import webapp2

from perf_insights import cloud_config


def _is_devserver():
  return os.environ.get('SERVER_SOFTWARE', '').startswith('Development')

_DEFAULT_MAPPER = """
<!DOCTYPE html>
<!--
Copyright (c) 2015 The Chromium Authors. All rights reserved.
Use of this source code is governed by a BSD-style license that can be
found in the LICENSE file.
-->
<link rel="import" href="/perf_insights/function_handle.html">
<link rel="import" href="/tracing/value/value.html">

<script>
tr.exportTo('pi.m', function() {

  function railMapFunction(results, runInfo, model) {
    results.addValue(new tr.v.DictValue(
        runInfo, 'userExpectationsCount',
        model.userModel.expectations.length));
  }
  pi.FunctionRegistry.register(railMapFunction);

  return {
    railMapFunction: railMapFunction
  };
});

</script>
"""

_DEFAULT_FUNCTION = 'railMapFunction'

_FORM_HTML = """
<!DOCTYPE html>
<html>
<body>
<form action="/cloud_mapper/create" method="POST">
Mapper: <br><textarea rows="50" cols="80" name="mapper">{mapper}</textarea>
<br>
FunctionName: <br><input type="text" name="mapper_function"
    value="{mapper_function}"/>
<br>
Query: <br><input type="text" name="query" value="{query}"/>
<br>
Corpus: <br><input type="text" name="corpus" value="{corpus}"/>
<br>
<input type="submit" name="submit" value="Submit"/>
</form>
</body>
</html>
"""


class TestPage(webapp2.RequestHandler):

  def get(self):
    form_html = _FORM_HTML.format(mapper=_DEFAULT_MAPPER,
                                  mapper_function=_DEFAULT_FUNCTION,
                                  query='MAX_TRACE_HANDLES=10',
                                  corpus=cloud_config.Get().default_corpus)
    self.response.out.write(form_html)

app = webapp2.WSGIApplication([('/cloud_mapper/test', TestPage)])
