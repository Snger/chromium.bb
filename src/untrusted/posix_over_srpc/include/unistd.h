#ifndef _UNTRUSTED_POSIX_OVER_SRPC_INCLUDE_UNISTD_H_
#define _UNTRUSTED_POSIX_OVER_SRPC_INCLUDE_UNISTD_H_

#include <sys/unistd.h>

char* getwd(char *__buf);

#endif  /* _UNTRUSTED_POSIX_OVER_SRPC_INCLUDE_UNISTD_H_ */
