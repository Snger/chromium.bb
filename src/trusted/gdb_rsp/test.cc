/*
 * Copyright 2010 The Native Client Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can
 * be found in the LICENSE file.
 */

#include <stdarg.h>
#include <stdio.h>
#include <stdlib.h>

#include "native_client/src/trusted/gdb_rsp/test.h"
#include "native_client/src/trusted/port/platform.h"

//  Mock portability objects
namespace port {
void IPlatform::LogInfo(const char *fmt, ...) {
  va_list argptr;
  va_start(argptr, fmt);

  vprintf(fmt, argptr);
}

void IPlatform::LogWarning(const char *fmt, ...) {
  va_list argptr;
  va_start(argptr, fmt);

  vprintf(fmt, argptr);
}

void IPlatform::LogError(const char *fmt, ...) {
  va_list argptr;
  va_start(argptr, fmt);

  vprintf(fmt, argptr);
}
}  // End of namespace port

int main(int argc, const char *argv[]) {
  int errs = 0;

  (void) argc;
  (void) argv;

  printf("Testing Utils.\n");
  errs += TestUtil();

  printf("Testing ABI.\n");
  errs += TestAbi();

  printf("Testing Packets.\n");
  errs += TestPacket();


  if (errs) printf("FAILED with %d errors.\n", errs);

  return errs;
}

