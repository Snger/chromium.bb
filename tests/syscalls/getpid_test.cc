/*
 * Copyright 2010 The Native Client Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can
 * be found in the LICENSE file.
 */


/*
 * Test for getpid syscall.
 */

#include "native_client/tests/syscalls/test.h"  //NOLINT

#include <errno.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

bool TestGetPid() {
  bool test_status;
  const char *testname = "getpid_test";
  pid_t pid_one = getpid();
  pid_t pid_two = getpid();

  // check if it's greater than 0.
  if ((pid_one > 0) && (pid_one == pid_two)) {
    test_status =
        test::Passed(testname,
                     "getpid returned what appears to be a valid pid.");
  } else if (pid_one == pid_two) {
    // a buffer size that should be sufficient
    char buffer[256];
    snprintf(buffer,
             255,
             "getpid returned an invalid pid for this test: %d",
             pid_one);
    buffer[255] = 0;
    test_status = test::Failed(testname, buffer);
  } else {
    char buffer[256];
    snprintf(buffer,
             255,
             "getpid returned different pids for the same process.  First "
             "time it was called: %d Second time: %d",
             pid_one, pid_two);
    buffer[255] = 0;
    test_status = test::Failed(testname, buffer);
  }
  return test_status;
}

/*
 * function testSuite()
 *
 *   Run through a complete sequence of file tests.
 *
 * returns true if all tests succeed.  false if one or more fail.
 */
bool TestSuite() {
  bool ret = true;
  ret &= TestGetPid();
  return ret;
}
