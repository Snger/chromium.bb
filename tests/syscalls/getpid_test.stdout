TEST PASSED: getpid_test: getpid returned what appears to be a valid pid.
All tests PASSED